MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -454,6 +481,16 @@ open class RouteLegProgress: NSObject {
 
         return currentClosest
     }
+    
+    /**
+     The waypoints remaining on the current leg, not including the leg’s destination.
+     */
+    func remainingWaypoints(among waypoints: [Waypoint]) -> [Waypoint] {
+        let legPolyline = Polyline(leg.coordinates)
+        let userCoordinateIndex = legPolyline.indexedCoordinateFromStart(distance: distanceTraveled)?.index ?? 0
+        // TODO: What if the index doesn’t monotonically increase due to backtracking?
+        return Array(waypoints.drop { legPolyline.closestCoordinate(to: $0.coordinate)?.index ?? 0 <= userCoordinateIndex })
```

Comment:
This would be more straightforward if Turf would provide a purely distance-based `indexedCoordinate(from:to:)`, or at least if IndexedCoordinate could indicate the distance from the start of the polyline instead of just the snapping distance: mapbox/turf-swift#78.